### DISCUSSION TITLE
errorMessage: `"w" parameter (width) of ${width} is not allowed`

### DISCUSSION THREAD

**Discussion by VegtMedia-Alex** [vercel/next.js#44978] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

"next": "^12.2.0",

### What browser are you using? (if relevant)

chrome

### Describe the Bug

ImageOptimizerCache class is blocking on the fly resizing.

### Expected Behavior

ImageOptimizerCache class should not block on the fly resizing.

### Link to reproduction

every version

### To Reproduce

I want more control over the src attribute in the next/future/image component. And at the same time want to use the image optimization from next and vercel.

So what i did is the following:

1) I created a custom loader for the image component

```
const loader = ({
	name,
	src,
	width,
}) => {
	return `${process.env.NEXT_PUBLIC_BASE_URL}/images/${name}?u=${src}&w=${width}`;
};
```

2) This way I can fully control the src prop url for example something like this:
http://localhost:4020/images/Some-Nice-Name.jpg?u=https://www.external.com/image.jpg&w=640

3) In the next.config.js I created a rewrite rule to a next api route

```
afterFiles: [
{
	'source': '/images/:path*',
	'destination': '/api/images/',
},
],
```
4) I created this API route to basicly proxy the /_next/image layer

```
import request from 'request';

export default async(req, res) => {
	try {
		if (req.method === 'GET' || req.method === 'OPTIONS') {
			let { u: url, w } = req.query;
			let imageBuffer = await request(`${process.env.NEXT_PUBLIC_BASE_URL}/_next/image/?url=${url}&w=${w}&q=75`);

			res.send(imageBuffer);
		} else {
			res.status(404).json({
				error: {
					code: 404,
					message: 'ROUTE_NOT_FOUND',
				},
			});
		}
	} catch (error) {
		res.status(500).json({
			error: JSON.stringify(error),
		});
	}
};
```

5) As soon as i send the request I get this error:

```
"w" parameter (width) of 640 is not allowed
```

6) When I go to this file: next/dist/server/image-optimizer and comment line 135 till 139

```
if (!sizes.includes(width)) {
    return {
        errorMessage: `"w" parameter (width) of ${width} is not allowed`
    };
}
```

7) Everything works as expected and you can easily build your own image server with the power of vercel and next?

Now my question is, is this blocked by design or maybe accept wildcard sizes in the next.config.js?

I think this would be a nice feature to have.

**Comment by NiedziolkaMichal** [comment]
@DB-Alex If there would be no width restrictions, somebody could overflow the image optimizer cache and drain resources by simply creating HTTP get requests with varying different `w` parameters.

**Reply by mledwards** [reply]
What are the width restrictions?
